src/tests/multihost/alltests/test_services.py

```diff
@@ -200,3 +202,98 @@ def test_0006_bz1909755(self, multihost, backupsssdconf):
+        process_id = (exceute_cmd(multihost, "pidof sssd").stdout_text.split()[0])
+        file_for_grep = f"/proc/{process_id}/environ"
+        grep_cmd = f'grep "PIZZA=YUMMY" {file_for_grep}'
+        cmd_check = multihost.client[0].run_command(grep_cmd, raiseonerr=False)
+        multihost.client[0].run_command('rm -f /etc/sysconfig/sssd', raiseonerr=False)
+        assert cmd_check.returncode == 0, "string 'PIZZA=YUMMY' not found in /proc file"
+
+    @pytest.mark.tier1_2
+    def test_0008_bz1516266(self, multihost, backupsssdconf):
+        """
+        :title: detailed debug and system-log message if krb5_init_context failed
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1516266
+        :id: fe126445-21d2-4572-b32a-b66c8486569f
+        :steps:
+          1. Update the sssd.conf with id and auth provider
+          2. Add 'includedir /var/lib/sss/pubconf/krb5.include.d/' in krb5.conf
+          3. Check sssd status, sssd should start
+          4. Revert back the krb.conf and add 'includedir /var/lib/sss/pubconf/krb5.include.d/$$$$' in krb5.conf
+             here $$$ is spaces after string
+          5. Check sssd status, sssd should fail to start
+        :expectedresults:
+          1. Successfully upadated sssd.conf
+          2. Successfully added string in krb5.conf
+          3. Successfully start the sssd service
+          4. Successfully reverted the krb5.conf and added string with spaces at the end
+          5. SSSD failing to start
+        """
+        # port bash to pytest
+        tools = sssdTools(multihost.client[0])
+        sssd_params = {'domains': 'LDAP',
+                       'debug_level': '9'}
+        tools.sssd_conf('sssd', sssd_params)
+        domain_section = 'domain/LDAP'
+        domain_params = {'id_provider': 'ldap',
+                         'ldap_uri': 'ldap://ldap.example.com',
+                         'ldap_search_base': 'dc=example,dc=com',
+                         'auth_provider': 'krb5',
+                         'krb5_server': 'kerberos.example.com',
+                         'krb5_realm': 'EXAMPLE.COM',
+                         'debug_level': '9'}
+        tools.sssd_conf(domain_section, domain_params)
+        # stop sssd, delete logs and cache, start sssd
+        tools.clear_sssd_cache()
+        random_file1 = 'random' + ''.join(random.choice(string.ascii_lowercase) for i in range(5))
+        take_bk = 'cp -f /etc/krb5.conf /etc/krb5.conf.backup'
+        multihost.client[0].run_command(take_bk, raiseonerr=False)
+        cmd_to_add = '{ echo "includedir /var/lib/sss/pubconf/krb5.include.d/"; cat /etc/krb5.conf; } > ' \
+                     f'/tmp/{random_file1}'
+        multihost.client[0].run_command(cmd_to_add, raiseonerr=False)
+        copy_radom = f'cp -f /tmp/{random_file1} /etc/krb5.conf'
+        multihost.client[0].run_command(copy_radom, raiseonerr=False)
+        start_sssd = multihost.client[0].service_sssd('restart')
+        assert start_sssd == 0
```

Comment:
If the test fails here, it will not restore krb5.conf and remove random file.
The assert could use a message explaining where exactly the test failed.

Reply:
updated.